pkg/controllers/dashboard/hostedcluster/controller.go

```diff
@@ -108,6 +132,28 @@ func (h handler) onClusterChange(key string, cluster *v3.Cluster) (*v3.Cluster,
 	return cluster, nil
 }
 
+func (h handler) removeLegacyOperatorIfExists(provider string) error {
+	systemProject, err := project.GetSystemProject(localCluster, h.projectCache)
+	if err != nil {
+		return err
+	}
+
+	systemProjectID := ref.Ref(systemProject)
+	_, systemProjectName := ref.Parse(systemProjectID)
+
+	legacyOperatorAppName := fmt.Sprintf(legacyOperatorAppNameFormat, provider)
+	_, err = h.apps.Get(systemProjectName, legacyOperatorAppName, metav1.GetOptions{})
```

Comment:
Can we `Delete` here and check for `IsNotFound` error?

Reply:
@thedadams I think the best thing to do here is probably to get it off the cache, is that okay with you? Will update.